tasks/main.yml

```diff
@@ -83,6 +94,25 @@
   when: docker_upstream|d() | bool
   tags: [ 'role::docker:config' ]
 
+- name: Configure Docker proxy
+  template:
+    src: 'etc/systemd/system/docker.service.d/http-proxy.conf.j2'
+    dest: '/etc/systemd/system/docker.service.d/http-proxy.conf'
+    owner: 'root'
+    group: 'root'
+    mode: '0644'
+  notify: [ 'Restart docker' ]
+  register: docker_register_systemd_service
+  when: (docker_upstream|d() | bool) and 
+    (inventory__environment is defined and inventory__environment.http_proxy is defined)
+  tags: [ 'role::docker:config' ]
+
+- name: Remove Docker proxy configuration
+  file:  path='/etc/systemd/system/docker.service.d/http-proxy.conf' state=absent
+  when: (docker_upstream|d() | bool) and 
+    (inventory__environment is not defined or inventory__environment.http_proxy is not defined)
+  tags: [ 'role::docker:config' ]
```

Comment:
This task also needs to be checked for changes to detect if `systemd` needs to be reloaded.